### ISSUE TITLE
Deepspeed strategy is slow for large model

### ISSUE THREAD

**Issue by feipenghe** [PyTorchLightning/pytorch-lightning#11901]
## 🐛 Bug

I fine-tune t5-large and try `ddp_sharded`, `ddp`,  `deepspeed_stage_3_offload` and `deepspeed_stage_3` strategies.
I find that  `deepspeed` strategies are five times slower than `ddp_sharded` and `ddp`. Also, when I use `deepspeed` CPU utilization close to 100% but power consumption is half of the full power consumption(150W w.r.t 300W). I think some parts of GPU are not used. 

`ddp` and `ddp_sharded` is around 1 iteration per second,
`deepspeed_stage_3` and `deepspeed_stage_3_offload` is around 5 second per iteration.

### Expected behavior

It's expected to have a similar training speed among these strategies.

### Environment

* CUDA:
	- GPU:
		- NVIDIA GeForce RTX 3090
		- NVIDIA GeForce RTX 3090
		- NVIDIA GeForce RTX 3090
		- NVIDIA GeForce RTX 3090
		- NVIDIA GeForce RTX 3090
		- NVIDIA GeForce RTX 3090
	- available:         True
	- version:           11.3
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.10.1+cu113
	- pytorch-lightning: 1.5.9
	- tqdm:              4.62.3
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.13
	- version:           #29~20.04.1-Ubuntu SMP Fri Jan 14 00:32:30 UTC 2022

### Additional context

This is the second problem I found in addition to https://github.com/PyTorchLightning/pytorch-lightning/issues/11760#issuecomment-1032409201.

**Comment by SeanNaren** [comment]
hey @feipenghe!

I think this is as expected; ddp_sharded (which is also the same theory of communication as deepspeed_stage_2) is much faster than deepspeed_stage_3 or any form of offloading.

`ddp_sharded`/`deepspeed_stage_2` - shards gradients/optimizer state across all your GPUs whilst remaining at parity with DDP
`deepspeed_stage_3` - shards parameters as well, increasing overall communication and thus being slower than DDP whilst reducing memory requirements
`deepspeed_stage_3_offload` or any offloading variant - offloads corresponding allocated memory to the CPU, meaning that we additionally have to factor in CPU -> GPU transfer times. 

Have you read our [documentation](https://pytorch-lightning.readthedocs.io/en/latest/advanced/advanced_gpu.html) page by any chance, this would also be beneficial for you!
